Update `artichoke` CLI to clap v4

lopopolo opened the pull request:
https://epage.github.io/blog/2022/09/clap4/

`--help` output is changed, so update the ui-test snapshots.

One of the neat new things in clap v4 is the ability to add a test to assert that the `clap::Command` is configured correctly. When a package has both a `lib.rs` and a binary target, I think `#[test]`s don't run in the `main.rs`, so this PR also takes the chance to migrate all of the `artichoke` clap CLI setup to a new `artichoke::ruby::cli` module.

lopopolo committed:
Update `artichoke` CLI to clap v4

https://epage.github.io/blog/2022/09/clap4/

`--help` output is changed, so update the ui-test snapshots.

epage commented on src/ruby/cli.rs:
```diff
@@ -0,0 +1,153 @@
+    // at this point is a broken pipe error, in which case, we want to ignore
+    // it and exit quietly.
+    //
+    // (This is the point of this helper function. clap's functionality for
+    // doing this will panic on a broken pipe error.)
```
I thought the broken pipe issue has been fixed for a while...

lopopolo replied:
I think it has been, but this code has worked the whole time and doesn't have that much maintenance cost for me to carry. I also prefer having visibility into all calls to `process::exit` instead of having them happen in a library if I can help it.

lopopolo replied:
But I can definitely clean up this comment since the clap side of things has been fixed.

lopopolo replied:
I just pushed a commit that removes this parenthetical comment. Thanks for catching that!

epage replied:
You can also use `err.print()` so you can control the exiting but get clap to do the printing, including colored output.  If you want more control over that and are fine with ANSI escape codes, you can now also call `err.render().ansi()`.

lopopolo replied:
Oh that's awesome!

## Before

<img width="779" alt="Screen Shot 2022-09-28 at 8 23 53 PM" src="https://user-images.githubusercontent.com/860434/192931725-35772305-9ba1-4e00-a051-e74bf1e06d4f.png">

## After

<img width="1042" alt="Screen Shot 2022-09-28 at 8 24 06 PM" src="https://user-images.githubusercontent.com/860434/192931703-cab299da-5157-4226-90f2-69b66a4ff25a.png">

epage replied:
Huh, noticed a couple issues in that.  Addressed in https://github.com/clap-rs/clap/pull/4287 which should be released soon

lopopolo replied:
Glad to be of help!

lopopolo committed:
Remove comment about clap panicking on broken pipes

This was fixed in 3.x.

Still keep the helper function, since it gives us control over the call to
process::exit and the exit code.

lopopolo committed:
Let clap print its own errors to get colorized output

lopopolo committed:
Use iter::once instead of an array iter

lopopolo commented:
Thanks for the code review @epage 🎉 ⚙️
